Utils/Dataflow/055_documents2TTL/documents2ttl.py

```diff
@@ -399,86 +401,66 @@ def fix_list_values(list_vals):
+    return True
 
 def main(argv):
     """
     Parsing command line arguments and processing JSON string from file or from stream
     :param argv: arguments
     :return:
     """
-    processor = stage.JSONProcessorStage()
-    processor.add_argument('-g', '--graph', action='store', type=str, nargs='?',
+    stage = pyDKB.dataflow.stage.JSON2TTLProcessorStage()
+    stage.add_argument('-g', '--graph', action='store', type=str, nargs='?',
                         help='Virtuoso DB graph name (default: %(default)s)',
                         default=GRAPH,
                         const=GRAPH,
                         metavar='GRAPH',
                         dest='GRAPH')
-    processor.add_argument('-O', '--ontology', action='store', type=str, nargs='?',
+    stage.add_argument('-O', '--ontology', action='store', type=str, nargs='?',
                         help='Virtuoso ontology prefix (default: %(default)s)',
                         default=ONTOLOGY,
                         const=ONTOLOGY,
                         metavar='ONT',
                         dest='ONTOLOGY')
-    processor.add_argument('-delim', '--delimiter', action='store', nargs='?',
-                        help=u'EOP marker for Kafka mode (default: \0)',
-                        default='',
-                        dest='EOPmarker')
-    processor.parse_args(argv)
-    define_globals(processor.ARGS)
-    gen = processor.input()
-
-    for f in gen:
-        data = f.content()
-        paper_id = data.get('dkbID')
-        doc_iri = get_document_iri(paper_id)
-        doc_ttl = ""
-        doc_ttl += '{docIRI} a <{ontology}#Paper> .\n' \
-            .format(docIRI=doc_iri, ontology=ONTOLOGY)
-        doc_ttl += document_glance(data.get('GLANCE'), doc_iri, PAPER_GLANCE_ATTRS)
-        doc_ttl += document_cds(data.get('CDS'), doc_iri, PAPER_CDS_ATTRS)
-
-        # supporting documents processing
-
-        if "supporting_notes" in data:
-            for note in data.get('supporting_notes'):
-                note_id = note.get('dkbID')
-                note_iri = get_document_iri(note_id)
-                doc_ttl += '{noteIRI} a <{ontology}#SupportingDocument> .\n' \
-                    .format(noteIRI=note_iri, ontology=ONTOLOGY)
-                doc_ttl += document_glance(note.get('GLANCE'), note_iri, NOTE_GLANCE_ATTRS)
-                doc_ttl += document_cds(note.get('CDS'), note_iri, NOTE_CDS_ATTRS)
-
-        doc_ttl += documents_links(data)
-
-        if processor.ARGS.mode == 'f':
-            if processor.ARGS.output_dir != '':
-                output = processor.ARGS.output_dir + '/' + paper_id + '.ttl'
-            else:
-                output = paper_id + '.ttl'
-            write_ttl2file(output, doc_ttl)
-        else:
-            sys.stdout.write(doc_ttl + processor.ARGS.EOPmarker)
-            sys.stdout.flush()
+    stage.parse_args(argv)
+    define_globals(stage.ARGS)
+    stage.process = process
+    stage.run()
+    stage.stop()
```

Comment:
Please check here: https://github.com/PanDAWMS/dkb/blob/pyDKB-refinement/Utils/Dataflow/test/pyDKB/json2TTL.py#L28

We want to run stop() even if the run() throws an exception, so try...finally is vital here.
Sorry I`ve misled everyone with example currently existing in pyDKB branch :(